server/api.js

```diff
@@ -3,7 +3,29 @@ import { Router } from "express";
 const router = Router();
 
 router.get("/", (_, res) => {
-	res.json({ message: "Hello, world!" });
+	res.json({ message: "Final Project!!!" });
+});
+
+const multer = require("multer");
+
+const fileStorageEngine = multer.diskStorage({
+	destination: (req, file, cb) => {
+		cb(null, "./images");
+	},
+	filename: (req, file, cb) => {
+		cb(null, Date.now() + "--" + file.originalname);
+	},
+});
+
+const upload = multer({ storage: fileStorageEngine });
+
+router.post("/upload", upload.single("image"), (req, res) => {
+	const title = req.body.title;
+	const author = req.body.author;
+
+	console.log(req.file);
+	console.log(title, author);
+	res.send("single file upload success");
 });
```

Comment:
based on what we did during node module and my research the  app.listen and app.use  is missing
app.listen(PORT,function(){
  console.log('Server is running at PORT:',PORT);
});
i found this link Using Express router with Node JS that might be helpful 
https://screenmaxx.in/using-express-router-with-node-js/#:~:text=The%20basic%20syntax%20of%20the%20express%20post%20route,req%20and%20save%20the%20data%20inside%20the%20database.

Reply:
Thank you Maira please check configurations, the template has many other files set up, therefore can't be compared to the exact method we learned in class, please read the extra files in the project, it is helpful to understand what they are doing and how they relate to each other.